src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationTask.kt

```diff
@@ -438,16 +443,16 @@ fun JvmCompilationTask.compileKotlin(
 
 /**
  * If any srcjars were provided expand the jars sources and create a new [JvmCompilationTask] with the
- * Java and Kotlin sources merged in.
+ * Java, Kotlin sources and META folder merged in.
  */
 internal fun JvmCompilationTask.expandWithSourceJarSources(): JvmCompilationTask =
   if (inputs.sourceJarsList.isEmpty()) {
     this
   } else {
     expandWithSources(
       SourceJarExtractor(
-        destDir = Paths.get(directories.temp).resolve("_srcjars"),
-        fileMatcher = IS_JVM_SOURCE_FILE,
+        destDir = Paths.get(directories.temp).resolve(SOURCE_JARS_DIR),
+        fileMatcher = { str: String -> IS_JVM_SOURCE_FILE.test(str) || "/$MANIFEST_DIR" in str },
```

Comment:
I'm not fully aware of the whole compilation process but if we add this logic here, it will also run for jars generated by `kapt` (as it's based on whole `input.sourceJarsList`). Since the issue for missing manifest files seems to be related to `ksp` only, could we possibly put this expand logic in a place where it wouldn't affect `kapt` generated jars? 💭

Reply:
@zalewskise I think we actually want this for Kapt as wellsince the meta directory could contain files needed in the final jar. One example that comes to mind is google-auto-service where it spits out information that's needed in the final jar.